iothub/device/src/IotHubModuleClient.cs

```diff
@@ -32,7 +32,7 @@ public class IotHubModuleClient : IotHubBaseClient
         private readonly SemaphoreSlim _moduleReceiveMessageSemaphore = new(1, 1);
 
         // Cloud-to-module message callback information
-        private volatile Tuple<Func<Message, object, Task<MessageResponse>>, object> _defaultEventCallback;
+        private volatile Tuple<Func<Message, object, Task<CloudToDeviceMessageAcknowledgement>>, object> _defaultEventCallback;
```

Comment:
I like CloudToDeviceMessageAcknowledgement for being descriptive of its function, but we're going to run into the same device vs module issue again. CloudToClient...?

Reply:
Good point...

Reply:
I'd rather revert to ```MessageAcknowledgement``` or ```MessageAcknowledgementType``` than use "Client" as a stand-in for both devices and modules. "Client" could mean device client, module client, or service client so it is less than helpful

Reply:
I'd vote for `MessageAcknowledgement` then

Reply:
Let's go with that then